src/nss-resolve/nss-resolve.c

```diff
@@ -393,6 +394,7 @@ enum nss_status _nss_resolve_gethostbyname3_r(
         _cleanup_(varlink_unrefp) Varlink *link = NULL;
         _cleanup_(json_variant_unrefp) JsonVariant *cparams = NULL;
         _cleanup_(resolve_hostname_reply_destroy) ResolveHostnameReply p = {};
+        _cleanup_free_ char *alias = NULL;
```

Comment:
hmm, wouldnÄ't `normalized_canonical` be a better name for this?

Reply:
This variable doesn't hold the canonical name though? It is assigned the normalized query name below.